packages/backend-function/package.json

```diff
@@ -32,7 +32,7 @@
     "uuid": "^9.0.1"
   },
   "peerDependencies": {
-    "aws-cdk-lib": "^2.158.0",
+    "aws-cdk-lib": "^2.168.0",
```

Comment:
Please update `aws-cdk-lib` and `aws-cdk` to matching version in all package.json files across the workspace.

Reply:
Updated in [d443b1f](https://github.com/aws-amplify/amplify-backend/pull/2269/commits/d443b1f0eaf3bfaf908d38ebf0e33b5434888208)

Reply:
I created a script `update-cdk-in-packages.js` to do this. Sharing in case it is useful to you in the future.

```js
/* eslint-disable */
const rootDir = process.cwd();
import fs from 'fs';
import path from 'path';

async function findPackageJsonFiles(dir) {
    let results = [];
    const list = fs.readdirSync(dir);
    for (let file of list) {
        file = path.resolve(dir, file);
        const stat = fs.statSync(file);
        if (stat && stat.isDirectory()) {
            results = results.concat(await findPackageJsonFiles(file));
        } else if (file.endsWith('package.json')) {
            const relativePath = path.relative(rootDir, file);
            if (/packages\/[\w-]+\/package\.json$/.test(relativePath)) {
                results.push(file);
            }
        }
    }
    return results;
}

function updatePackageVersion(packageJson, packageName, newVersion) {
    if (packageJson.dependencies && packageJson.dependencies[packageName]) {
        packageJson.dependencies[packageName] = newVersion;
    }
    if (packageJson.devDependencies && packageJson.devDependencies[packageName]) {
        packageJson.devDependencies[packageName] = newVersion;
    }
    if (packageJson.peerDependencies && packageJson.peerDependencies[packageName]) {
        packageJson.peerDependencies[packageName] = newVersion;
    }
}

async function updateCdkVersion(filePaths) {
    for (let filePath of filePaths) {
        const packageJson = JSON.parse(fs.readFileSync(filePath, 'utf8'));
        updatePackageVersion(packageJson, 'aws-cdk-lib', '^2.168.0');
        updatePackageVersion(packageJson, 'aws-cdk', '^2.168.0');
        fs.writeFileSync(filePath, JSON.stringify(packageJson, null, 2) + `\n`, 'utf8');
    }
}

async function main() {
    const packageJsonFiles = await findPackageJsonFiles(path.join(rootDir, 'packages'));
    await updateCdkVersion(packageJsonFiles);
    console.log('Updated package.json files:', packageJsonFiles);
}

main();
/* eslint-enable */
```

Reply:
thank you.
there will be one more thing needed - additional changeset, since most of packages got touched.
Please see https://github.com/aws-amplify/amplify-backend/commit/b56d344e27f48ba235644fb695e6de465028bc65 for reference. This commit contains a changeset file from last time we upgraded CDK.

Reply:
Thanks for the reference, that helps.